### PR TITLE
[stable/jenkins] increase retries for sidecar reload

### PR DESCRIPTION
**Pull request by torstenwalter**
#### What this PR does / why we need it:

This is an attempt to fix ci build errors fro Jenkins chart.

```
I0614 18:14:07.070] [2020-06-14 18:14:04] Selected resource type: ('configmap',)
I0614 18:14:07.071] [2020-06-14 18:14:04] Config for cluster api loaded...
I0614 18:14:07.071] [2020-06-14 18:14:04] Unique filenames will not be enforced.
I0614 18:14:07.072] [2020-06-14 18:14:04] Working on configmap jenkins-presubmit-22641-1272230382288244742-i02ztn8t8l/jenkins-i02ztn8t8l-jenkins-jcasc-config
I0614 18:14:07.072] [2020-06-14 18:14:04] File in configmap jcasc-default-config.yaml ADDED
I0614 18:14:07.073] 2020-06-14 18:14:04,907 WARNING Retrying (Retry(total=4, connect=4, read=5, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fc06091ad10>: Failed to establish a new connection: [Errno 111] Connection refused')': /reload-configuration-as-code/?casc-reload-token=jenkins-i02ztn8t8l-85c5968964-vz5fq
I0614 18:14:07.073] 2020-06-14 18:14:05,308 WARNING Retrying (Retry(total=3, connect=3, read=5, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fc06092b3d0>: Failed to establish a new connection: [Errno 111] Connection refused')': /reload-configuration-as-code/?casc-reload-token=jenkins-i02ztn8t8l-85c5968964-vz5fq
I0614 18:14:07.074] 2020-06-14 18:14:06,109 WARNING Retrying (Retry(total=2, connect=2, read=5, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fc06092b8d0>: Failed to establish a new connection: [Errno 111] Connection refused')': /reload-configuration-as-code/?casc-reload-token=jenkins-i02ztn8t8l-85c5968964-vz5fq
```

As you can see from the logs the sidecar container is trying to send a reload configuration as code request, but Jenkins is not fully started yet. I am increasing the `REQ_RETRY_CONNECT` as documented in https://github.com/kiwigrid/k8s-sidecar.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by wmcdona89** [comment]
Ah...thanks for investigating and fixing! 

/lgtm

File: stable/jenkins/values.yaml [hunk 1]
```diff
@@ -308,7 +308,9 @@ master:
         #   requests:
         #     cpu: 50m
         #     memory: 50Mi
-      # env:
+      env:
+        - name: REQ_RETRY_CONNECT
+          value: 10
```
Comment (wmcdona89) [inline stable/jenkins/values.yaml]: @torstenwalter looks like this needs to be in quotes as CI is failing. could also update the default value in the README.

Comment (torstenwalter) [reply stable/jenkins/values.yaml]: We know in a couple of minutes if this fixes it.

**Comment by torstenwalter** [comment]
@wmcdona89 Any idea why tests are failing?

**Comment by torstenwalter** [comment]
The error is always:
```
I0615 09:13:39.525] E0615 09:13:39.506976     193 portforward.go:233] lost connection to pod
214
I0615 09:13:39.525] Error: transport is closing
```

Not sure where this is coming from. in one of the previous logs I wasn't sure if the init container had already finished. So I added some debug output.

This build was already killed before any of the init container output was shown:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/helm_charts/22787/pull-charts-e2e/1272456875333390339/

So it's probably a totally unrelated issue.

**Comment by scottrigby** [comment]
Testing infra was fixed, so last failure shouldn't happen again.
But version needs another bump first, since #22641 was just merged

**Comment by scottrigby** [comment]
Try retest after resolving merge conflict

**Comment by torstenwalter** [comment]
@scottrigby Many thanks for fixing the test infra!!!

**Pull request merged**